File: frontend/src/types/report.ts
import { ReportReasonCode, ReportStatus } from './enums';

export type { ReportReasonCode, ReportStatus };

export interface ReportDto {
  id: string;
  reporterUserId: string;
  reporterEmail?: string;
  targetType: string;
  targetId: string;
  reason?: string;
  reasonCode?: ReportReasonCode;
  description?: string;
  evidence?: string;
  status: ReportStatus;
  assignedModeratorId?: string;
  resolution?: string;
  createdAt: string;
  resolvedAt?: string;
}

export interface CreateReportRequest {
  targetType: string;
  targetId: string;
  reasonCode: ReportReasonCode;
  description: string;
  evidence?: string;
}

export interface ReportTargetContextDto {
  targetType: string;
  targetId: string;
  exists: boolean;
  canReport: boolean;
  displayName?: string | null;
  ownerEmail?: string | null;
  reason?: string | null;
}

export interface ReportActionDto {
  id: string;
  reportId: string;
  actorUserId: string;
  action: string;
  reason: string;
  createdAt: string;
}

export interface ReportDetailDto {
  report: ReportDto;
  actions: ReportActionDto[];
}

export interface ReportListResponse {
  items: ReportDto[];
  total: number;
  page: number;
  pageSize: number;
}

File: src/Domain/Entities/Report.cs
using StartupConnect.Domain.Enums;

namespace StartupConnect.Domain.Entities;

public sealed class Report : BaseEntity
{
    public Guid ReporterUserId { get; set; }

    public User ReporterUser { get; set; } = null!;

    public string TargetType { get; set; } = string.Empty;

    public Guid TargetId { get; set; }

    public string Reason { get; set; } = string.Empty;

    public ReportReasonCode ReasonCode { get; set; } = ReportReasonCode.Other;

    public string Description { get; set; } = string.Empty;

    public string? Evidence { get; set; }

    public ReportStatus Status { get; set; } = ReportStatus.Pending;

    public Guid? AssignedModeratorId { get; set; }

    public User? AssignedModerator { get; set; }

    public string? Resolution { get; set; }

    public DateTimeOffset? ResolvedAt { get; set; }
}
